fix entry undo and redo don't trigger OnChanged

mbaklor opened the pull request:
### Description:
using undo and redo on an entry widget didn't trigger OnChanged, so I added a call to the callback in both methods.

Fixes #5710 

### Checklist:
<!-- Please tick these as appropriate using [x] -->

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

andydotxyz commented on widget/entry.go:
```diff
@@ -426,6 +426,10 @@ func (e *Entry) Redo() {
 	e.updateText(newText, false)
 	e.CursorRow, e.CursorColumn = e.rowColFromTextPos(pos)
 	e.syncSelectable()
+	cb := e.OnChanged
```
This could be simplified:

```go
	if cb := e.OnChanged; cb != nil {
```

mbaklor replied:
this is fair but I wanted to keep it in line with the other places we call the callback, I'll fix when I get home, we can remove the whole `cb := ` at this point since we no longer have to mutex lock

andydotxyz replied:
I did do that check to make sure it wasn't suitably matching - but in the other cases the callback was taken a long time before the callback. Perhaps they do indeed all need fixed with the mutex changes, not sure.

andydotxyz commented on widget/entry.go:
```diff
@@ -426,6 +426,10 @@ func (e *Entry) Redo() {
 	e.syncSelectable()
+	cb := e.OnChanged
+	if cb != nil {
+		cb(newText) // We know that the text has changed.
```
Avoid comments unless it is required and cannot be communicated through cleaner code - this seems self-evident so no need

mbaklor replied:
oops I was sure I removed them that's my bad, I'll fix when I get home 😅

mbaklor commented:
Fixed the stylistic things, no need to copy the callback now that there isn't a mutex lock, and removed the comment that I accidentally yanked from elsewhere

andydotxyz reviewed (approved):
Thanks!
